TPP.ArgsParsing/TypeParsers/ManyOfParser.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using TPP.ArgsParsing.Types;
+
+namespace TPP.ArgsParsing.TypeParsers
+{
+    /// <summary>
+    /// A parser that parses values using the generic type argument of <see cref="ManyOf{T}"/>,
+    /// which essentially represent a list of homogeneously typed values.
+    /// Parsing never fails, but the resulting list may be empty.
+    /// <br/>
+    /// The parser has a few restrictions:
+    /// <ul>
+    ///   <li>
+    ///   It is assumed that each list entry consumes at least one argument, so it will not work with e.g. Optional.
+    ///   </li>
+    ///   <li>
+    ///   It just greedily parses as many arguments as possible. When used in a way where parsing may be ambiguous to
+    ///   any subsequent types, the parser will consume arguments that were meant for that subsequent type.
+    ///   Since there is no backtracking for successes, overall parsing will then fail because parsing the subsequent
```

Comment:
might be worth implementing at the cost of slowing down parsing of commands that use ManyOf